src/ImmersedBoundaries/mask_immersed_field.jl

```diff
@@ -0,0 +1,15 @@
+using KernelAbstractions
+using Oceananigans.Architectures: architecture, device_event
+
+mask_immersed_field!(field, ib) = NoneEvent()
+
+function mask_immersed_field!(field::AbstractField{X, Y, Z}, ib::GridFittedImmersedBoundary) where {X, Y, Z}
+    arch = architecture(field)
+    loc = (X(), Y(), Z())
+    return launch!(arch, field.grid, :xyz, _mask_immersed_field!, field, loc, ib; dependencies = device_event(arch))
+end
+
+@kernel function _mask_immersed_field!(field, loc, ib)
+    i, j, k = @index(Global, NTuple)
+    @inbounds field[i, j, k] = ifelse(solid_node(loc..., i, j, k, ib.grid, ib), 0, field[i, j, k])
+end
```

Comment:
I guess this is the only part that needs to be modified for `GridFittedImmersedBoundary` to work with `IncompressibleModel`?

Since modifying the velocities would cause the velocity field to become divergent, you'd want to relax the velocity instead like in https://github.com/CliMA/Oceananigans.jl/blob/57ac2d2cf16f08218037376c7af832ec457940e7/src/Models/IncompressibleModels/correct_incompressible_immersed_tendencies.jl#L43-L65

Reply:
For `IncompressibleModel` you make a choice: either 1) the boundary pressure gradient / boundary conditions are wrong, or 2) the velocity field is divergent.

I don't think relaxation solves the issue --- but perhaps your hypothesis is that is mitigates the problem? I'm not sure. Either way, tracer is not conserved.

The tracer conservation issue can be _solved_ by eliding advective fluxes in addition to diffusive fluxes on the boundary.

The larger issue can be solved by modifying the pressure solver. Perhaps the `PreconditionedConjugateGradientSolver` to "polish off" the pressure field obtained naively without considering boundary conditions, or we can design an iterative scheme with the FFT based pressure solver.

Reply:
Ah I misunderstood this comment originally. I believe the quoted function in `correct_incompressible_immersed_tendencies.jl` is _identical_ to `mask_immersed_fields.jl`. I believe the correction just calculates `Gⁿ` such that the RK update masks the velocity to 0. I believe `mask_immersed_fields!` is just a more straightforward way of achieving the same thing. So the current implementation doesn't implement relaxation.

Reply:
So I guess with the functionality provided by this PR you could achieve identical functionality as previously (except with no flux boundary conditions) by applying `mask_immersed_fields!` to the predictor velocity rather than in `update_state!`.

In that case tracer leaks across the boundary. But this can be fixed by eliding advective fluxes in addition to diffusive fluxes.

It'd be worth investigating whether masking of all fields in `update_state!` _in addition to_ masking of the predictor velocities prior to the pressure solve results in a non-divergent velocity field that satisfies boundary conditions.